File: src/main/java/com/github/pareronia/i18n_puzzles/Puzzle2025_05.java
package com.github.pareronia.i18n_puzzles;

import java.util.List;
import java.util.stream.IntStream;

import com.github.pareronia.i18n_puzzles.common.SolutionBase;

public class Puzzle2025_05 extends SolutionBase<Integer>{

    private static final int POO = "💩".codePointAt(0);

    private Puzzle2025_05(final boolean debug) {
        super(debug);
    }

    public static Puzzle2025_05 create() {
        return new Puzzle2025_05(false);
    }

    public static Puzzle2025_05 createDebug() {
        return new Puzzle2025_05(true);
    }

    @Override
    public Integer solve(final List<String> input) {
        return (int) IntStream.range(0, input.size())
            .filter(i -> {
                final int[] cps = input.get(i).codePoints().toArray();
                return cps[(i * 2) % cps.length] == POO;
            })
            .count();
    }

    @Override
    public void samples() {
        final List<String> input = List.of(
                " ⚘   ⚘ ",
                "  ⸫   ⸫",
                "🌲   💩  ",
                "     ⸫⸫",
                " 🐇    💩",
                "⸫    ⸫ ",
                "⚘🌲 ⸫  🌲",
                "⸫    🐕 ",
                "  ⚘  ⸫ ",
                "⚘⸫⸫   ⸫",
                "  ⚘⸫   ",
                " 💩  ⸫  ",
                "     ⸫⸫"
        );
        assert Puzzle2025_05.createDebug().solve(input) == 2;
    }

    public static void main(final String[] args) throws Exception {
        Puzzle2025_05.create().run();
    }
}

File: src/main/python/Puzzle2025_05.py
import sys

from i18n_puzzles.common import InputData
from i18n_puzzles.common import SolutionBase

TEST = (
    " ⚘   ⚘ ",
    "  ⸫   ⸫",
    "🌲   💩  ",
    "     ⸫⸫",
    " 🐇    💩",
    "⸫    ⸫ ",
    "⚘🌲 ⸫  🌲",
    "⸫    🐕 ",
    "  ⚘  ⸫ ",
    "⚘⸫⸫   ⸫",
    "  ⚘⸫   ",
    " 💩  ⸫  ",
    "     ⸫⸫",
)

Output = int
POO = "💩"


class Solution(SolutionBase[Output]):
    def solve(self, input: InputData) -> Output:
        return sum(
            line[(i * 2) % len(line)] == POO for i, line in enumerate(input)
        )

    def samples(self) -> None:
        assert self.solve(TEST) == 2


solution = Solution(5)


def main() -> None:
    solution.run(sys.argv)


if __name__ == "__main__":
    main()
